packages/nimble-components/src/table/models/table-validator.ts

```diff
@@ -45,21 +47,30 @@ export class TableValidator<TData extends TableNode> {
         );
     }
 
-    public validateSelectionMode(
+    public validateIdFieldConfiguration(
         selectionMode: TableRowSelectionMode,
-        idFieldName: string | undefined
+        idFieldName: string | undefined,
+        parentIdFieldName: string | undefined
     ): boolean {
         if (selectionMode === TableRowSelectionMode.none) {
             this.idFieldNameNotConfigured = false;
         } else {
             this.idFieldNameNotConfigured = typeof idFieldName !== 'string';
         }
 
+        if (
+            !this.idFieldNameNotConfigured
+            && parentIdFieldName !== undefined
+            && idFieldName === undefined
+        ) {
+            this.idFieldNameNotConfigured = true;
+        }
+
         return !this.idFieldNameNotConfigured;
```

Comment:
What would you think about rewriting this function a bit to make it more clear what's going on? Maybe something like this:

```ts
const idFieldNameRequired = selectionMode !== TableRowSelectionMode.none || typeof parentIdFieldName !== 'string';
if (idFieldNameRequired) {
    this.idFieldNameNotConfigured = typeof idFieldName !== 'string';
} else {
    this.idFieldNameNotConfigured = false;
}
return !this.idFieldNameNotConfigured;
```